### PR TITLE
Refactor zoo of read/write, load/store StableMem operations to share more code.

### PR DESCRIPTION
**Pull request by crusso**

### CODE REVIEW

**Comment by crusso** [comment]
@ggreif I decide to refactor all those (unguarded) read/write and guarded load/store operations to factor out the common bits. See what you think and I'll merge it in to the parent branch.

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -3805,147 +3805,63 @@ module StableMem = struct
...
-        ("offset", I32Type) [I32Type]
-        (fun env get_offset ->
-          Stack.with_words env "temp_ptr" 1l (fun get_temp_ptr ->
-            get_temp_ptr ^^ get_offset ^^ compile_unboxed_const 1l ^^
-            IC.system_call env "ic0" "stable_read" ^^
-            get_temp_ptr ^^
-            G.i (Load {ty = I32Type; align = 0; offset = 0l; sz = Some Wasm.Types.(Pack8, ZX)})))
-    | _ -> assert false
+  let read_word32 env =
+    read env "word32" I32Type 4l load_unskewed_ptr
```
Comment (ggreif) [inline src/codegen/compile.ml:3836]: The common usage pattern is `read env "thingy4xN" ⟨wasmtype⟩ N ⟨dest⟩`, you can exploit that if you want...

**Review by ggreif — APPROVED** [review]
Abstraction FTW!

File: src/codegen/compile.ml (hunk 1 above)
Comment (crusso) [reply src/codegen/compile.ml:3836]: Life is too short man. 

Was wondering about about pulling the range_check in to avoid the nested function call...

**Pull request closed without merging**